modules/evm/src/lib.rs

```diff
@@ -117,26 +122,40 @@ pub trait Trait: frame_system::Trait + pallet_timestamp::Trait {
 pub const STORAGE_SIZE: u32 = 64;
 
 #[derive(Clone, Eq, PartialEq, RuntimeDebug, Encode, Decode)]
-pub struct ContractInfo {
+pub struct ContractInfo<T: Trait> {
 	pub storage_count: u32,
 	pub code_hash: H256,
+	pub existential_deposit: BalanceOf<T>,
+	pub maintainer: T::AccountId,
 }
 
-impl ContractInfo {
+impl<T: Trait> ContractInfo<T> {
 	pub fn total_storage_size(&self) -> u32 {
 		self.storage_count.saturating_mul(STORAGE_SIZE)
 	}
 }
 
 #[derive(Clone, Eq, PartialEq, RuntimeDebug, Encode, Decode)]
-pub struct AccountInfo<Index> {
-	pub nonce: Index,
-	pub contract_info: Option<ContractInfo>,
+pub struct AccountInfo<T: Trait> {
+	pub nonce: T::Index,
+	pub contract_info: Option<ContractInfo<T>>,
+	pub storage_rent_deposit: BalanceOf<T>,
+	pub storage_quota: u32,
+	/// If is a contract, the storage_usage is additional storage from
+	/// maintainer. If is a maintainer, the storage_usage is the sum of
+	/// additional storage required by all contracts.
+	pub storage_usage: u32,
```

Comment:
contract can also be maintainer. I think we should just have two fields. `additional_storage_size` and `storage_quota_usage`